runtime/sema/access.go

```diff
@@ -32,6 +32,7 @@ import (
 
 type Access interface {
 	isAccess()
+	IsPrimitiveAccess() bool
```

Comment:
Could call-sites simply check if a given `var access Access` is a primitive access via `_, ok := access.(PrimitiveAccess)` instead? It looks like implementations of the function are simply true or false based on the type being `PrimitiveAccess` or not

Reply:
They could, but this makes it easier to pass to a function, or to otherwise use without having to declare a new variable. It's just personally preference, if you feel strongly I can remove this.